flaml/model.py

```diff
@@ -443,95 +494,121 @@ def _preprocess(self, X, y=None, **kwargs):
                 X=X,
                 Y=y,
                 task=self._task,
-                hf_args=self.hf_args,
+                hf_args=self._training_args,
                 tokenizer=self._tokenizer,
             )
         else:
             return X, None
 
-    def _model_init(self, num_labels, per_model_config):
+    def _model_init(self):
         from .nlp.utils import load_model
 
         this_model = load_model(
-            checkpoint_path=self.hf_args.model_path,
+            checkpoint_path=self._training_args.model_path,
             task=self._task,
-            num_labels=num_labels,
-            per_model_config=per_model_config,
+            num_labels=self._num_labels,
         )
         return this_model
 
-    def _get_training_args(self, local_rank=-1):
-        import transformers
+    def preprocess_data(self, X, y):
+        from datasets import Dataset
 
-        if self._task in NLG_TASKS:
-            self._training_args_config["predict_with_generate"] = True
-
-        if transformers.__version__.startswith("3"):
-            training_args = self._TrainingArguments(
-                report_to=[],
-                output_dir=self._trial_dir,
-                do_train=True,
-                do_eval=True,
-                eval_steps=self._ckpt_freq,
-                evaluate_during_training=True,
-                save_steps=self._ckpt_freq,
-                logging_steps=self._ckpt_freq,
-                save_total_limit=0,
-                metric_for_best_model="loss",
-                fp16=self.hf_args.fp16
-                if self._kwargs.get("gpu_per_trial") > 0
-                else False,
-                no_cuda=True if self._kwargs.get("gpu_per_trial") == 0 else False,
-                local_rank=local_rank,
-                per_device_eval_batch_size=self.hf_args.per_device_eval_batch_size,
-                **self._training_args_config,
+        if (self._task not in NLG_TASKS) and (self._task != TOKENCLASSIFICATION):
+            processed_X, _ = self._preprocess(X=X, **self._kwargs)
+            processed_y = y
+        else:
+            processed_X, processed_y = self._preprocess(X=X, y=y, **self._kwargs)
+
+        processed_dataset = Dataset.from_pandas(
+            TransformersEstimator._join(processed_X, processed_y)
+        )
+        return processed_dataset, processed_X, processed_y
+
+    @property
+    def _num_labels(self):

```

Comment:
property is usually public, so don't start with '_'. I could be wrong. If so, please find me some evidence.